src/features/defi/components/Deposit/Deposit.tsx

```diff
@@ -144,20 +146,33 @@ export const Deposit = ({
   )
 
   const handlePercentClick = useCallback(
-    (percent: number) => {
-      if (onPercentClick) onPercentClick(percent)
-      // The human crypto amount as a result of amount * percentage / 100, possibly with too many digits
-      const percentageCryptoAmount = bnOrZero(cryptoAmountAvailable).times(percent)
-      const percentageFiatAmount = percentageCryptoAmount.times(marketData.price)
+    async (percent: number) => {
+      const { percentageCryptoAmount, percentageFiatAmount } = await (async () => {
+        if (onPercentClick) {
+          const { amountCryptoPrecision: _amountCryptoPrecision, fiatAmount: _fiatAmount } =
+            await onPercentClick(percent)
+          return {
+            percentageCryptoAmount: bn(_amountCryptoPrecision),
+            percentageFiatAmount: _fiatAmount,
+          }
+        }
+
+        const _percentageCryptoAmount = bnOrZero(cryptoAmountAvailable).times(percent)
+        const _percentageFiatAmount = _percentageCryptoAmount.times(marketData.price)
+
+        return {
+          percentageCryptoAmount: _percentageCryptoAmount,
+          percentageFiatAmount: _percentageFiatAmount,
+        }
+      })()
+
       const percentageCryptoAmountHuman = percentageCryptoAmount
         .decimalPlaces(asset.precision)
         .toString()
```

Comment:
if you do decide to keep this here, I always used `toFixed` over `toString()` for any crypto amounts

Reply:
display only for sure since we truncate anyways, but even with `decimalPlaces`, `toString()` can result in scientific notation format if the number is big enough (unlikely, but the possibility always exists with `toString()`)